nestkernel/sonata_connector.cpp

```diff
@@ -563,13 +545,14 @@ SonataConnector::read_subset_( const H5::DataSet& dataset,
 }
 
 void
-SonataConnector::create_edge_type_id_2_syn_spec_( DictionaryDatum edge_params )
+SonataConnector::create_edge_type_id_2_syn_spec_( Dictionary edge_params )
 {
-  for ( auto it = edge_params->begin(); it != edge_params->end(); ++it )
+  for ( auto& syn_kv_pair : edge_params )
```

Comment:
Maybe here `for ( auto& [k_syn, v_syn] : edge_params )` and the corresponding changes in the loop?

Reply:
I see!